spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/web/jetty/JettyMetricsAutoConfiguration.java

```diff
@@ -50,4 +55,18 @@ public JettyServerThreadPoolMetricsBinder jettyServerThreadPoolMetricsBinder(Met
 		return new JettyServerThreadPoolMetricsBinder(meterRegistry);
 	}
 
+	@Bean
+	@ConditionalOnBean(MeterRegistry.class)
+	@ConditionalOnMissingBean({ JettyConnectionMetrics.class, JettyConnectionMetricsBinder.class })
+	public JettyConnectionMetricsBinder jettyConnectionMetricsBinder(MeterRegistry meterRegistry) {
+		return new JettyConnectionMetricsBinder(meterRegistry);
+	}
+
+	@Bean
+	@ConditionalOnBean(MeterRegistry.class)
+	@ConditionalOnMissingBean({ JettySslHandshakeMetrics.class, JettySslHandshakeMetricsBinder.class })
+	public JettySslHandshakeMetricsBinder jettySslHandshakeMetricsBinder(MeterRegistry meterRegistry) {
```

Comment:
Is there some other Ssl-related bean we could use to improve this conditional guard on (aka only load when actually using SSL)?

Reply:
We have the `server.ssl.enabled` property. It's possible that SSL may have been enabled through manual configuration, but perhaps it's reasonable to lose auto-configuration of SSL-related metrics in that case.